### DISCUSSION TITLE
entproto: grpc-gateway

### DISCUSSION THREAD

**Discussion by andybritton** [ent/ent#1671] [Ideas]
Reading through the grpc implementation I can't see anything about adding additional options to the service such as grpc-gateway/google annotations. Is this something that's being considered at all? How would modifying the generated .proto's be effected in future generate commands?

**Comment by rotemtam** [comment]
Hi @andybritton,

Thanks for reaching out!

As with any generated code, as the comment on the top of the file states:
```protobuf
// Code generated by entproto. DO NOT EDIT.
```
Whatever changes you make to the generated .proto file will be overridden the next time you generate the code. 

Addressing your questions, for "out of the box" proto options such as `java_package` etc. the implementation path is very clear and we will definitely do it. 

For things that are more complex such as configuring gRPC gateway through ent annotations, we would need to come up with a good API. I haven't put much thought into that yet and would love to hear your thoughts.  How would such an API look like on the ent schema? 

Off the top of my head, as a workaround, I would create an additional `.proto` file in the ent/proto/entpb directory. In the file, create a new service, _preserving_ the same interface of the generated service, and use the generated service as an implementation of your service. This should keep working unless we change the generated interface of the service definition (which is highly unlikely to happen). 

WDYT?
